### PR TITLE
feat: LEAP-1469: Implement universal linking modes

### PR DESCRIPTION
**Pull request by Gondragos**
This PR provides the system of universal linking modes. It's the ideological heir of `Relation Mode` with reusing a lot of its own code with a couple of renamings and refactoring. 
As the result we have separated `RelationMode` and `CommentMode`. The second one will provide ability to link comments to regions.

### PR fulfills these requirements
- [x] Commit message(s) and PR title follows the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
- [x] Tests for the changes have been added/updated (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [x] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [x] Self-reviewed and ran all changes on a local instance (for bug fixes/features)

#### Change has impacts in these area(s)
- [ ] Product design
- [ ] Backend (Database)
- [ ] Backend (API)
- [x] Frontend

### Describe the reason for change
We need a new mode mostly simmilar to `RelationMode`. That causes refactoring to reuse most of the functionality.

#### What feature flags were used to cover this change?
We avoided to using feature flags here as it might not affect the existing functionality.

### Does this PR introduce a breaking change?
- [ ] Yes, and covered entirely by feature flag(s)
- [ ] Yes, and covered partially by feature flag(s)
- [ ] No
- [x] Not sure (briefly explain the situation below)

It can if there are some problems with code 😅

### What level of testing was included in the change?
_(check all that apply)_
- [ ] e2e
- [x] integration
- [ ] unit

### Which logical domain(s) does this change affect?
`LinkingModes`, `RelationMode`, `CommentMode`

### CODE REVIEW

File: web/libs/editor/src/stores/Annotation/LinkingModes.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+      addLinkedRegion(region) {
+        if (self.currentLinkingMode) {
+          self.currentLinkingMode.addLinkedRegion?.(region);
+        }
+      },
+      // @deprecated
+      startRelationMode(obj) {
+        console.warn(
+          "`startRelationMode` is deprecated. Use `startLinkingMode` with `CREATE_RELATION_MODE` as first instead.",
```
Comment (hlomzik) [inline web/libs/editor/src/stores/Annotation/LinkingModes.js]:
```suggestion
          "`startRelationMode` is deprecated. Use `startLinkingMode(CREATE_RELATION_MODE, obj)` instead.",
```

File: web/libs/editor/src/stores/Annotation/LinkingModes.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+        self.linkingMode = false;
+      },
+
+      addLinkedRegion(region) {
+        if (self.currentLinkingMode) {
+          self.currentLinkingMode.addLinkedRegion?.(region);
+        }
+      },
+      // @deprecated
```
Comment (hlomzik) [inline web/libs/editor/src/stores/Annotation/LinkingModes.js]:
```suggestion
      // @deprecated Use `startLinkingMode(CREATE_RELATION_MODE, obj)` instead
```

File: web/libs/editor/src/stores/Annotation/LinkingModes.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+      startRelationMode(obj) {
+        console.warn(
+          "`startRelationMode` is deprecated. Use `startLinkingMode` with `CREATE_RELATION_MODE` as first instead.",
+        );
+        self.startLinkingMode(RelationMode.key, obj);
+      },
+      // @deprecated
```
Comment (hlomzik) [inline web/libs/editor/src/stores/Annotation/LinkingModes.js]:
```suggestion
      // @deprecated Use `stopLinkingMode` instead
```

File: web/libs/editor/src/stores/Annotation/LinkingModes.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    get relationMode() {
+      console.warn("`relationMode` is deprecated. Use `isLinkingMode` instead.");
+      return self.isLinkingMode;
+    },
+  }))
+  .actions((self) => {
+    return {
+      afterCreate() {
+        self.linkingModes.set(RelationMode.key, RelationMode.model.create({}));
+        self.linkingModes.set(CommentMode.key, CommentMode.model.create({}));
```
Comment (hlomzik) [inline web/libs/editor/src/stores/Annotation/LinkingModes.js]: Can we declare them statically in `model()`?

File: web/libs/editor/tests/integration/data/linked_modes/to_coments.ts
Comment (hlomzik) [inline web/libs/editor/tests/integration/data/linked_modes/to_coments.ts:1]: `to_comments` — one `m` is missing

File: web/libs/editor/src/stores/Annotation/LinkingModes.js (hunk 4 above)
Comment (Gondragos) [reply web/libs/editor/src/stores/Annotation/LinkingModes.js]: I afraid that they could be rewritten as they live in the `model` and all fields there are affected by the data we provide on creating the store.

Comment (hlomzik) [reply web/libs/editor/src/stores/Annotation/LinkingModes.js]: that's definitely something we should not worry about. on a level of Annotation store we connect all other store parts via model and they are fully controlled by us (in contrary to tag models for example, which are instantiated from config). so better to move this to model for consistency and readability

File: web/libs/editor/src/tags/object/RichText/model.js [hunk 5]
```diff
@@ -413,7 +413,7 @@ const Model = types
         self.regs.forEach((r) => r.setHighlight(false));
         if (!region) return;
 
-        if (region.annotation.relationMode) {
+        if (region.annotation.isLinkingMode) {
           region.setHighlight(true);
```
[the remark this answers is not in the document]
Comment (Gondragos) [reply web/libs/editor/src/tags/object/RichText/model.js:417]: Done. Is it ok now?

**Review by hlomzik — APPROVED** [review]

**Review by MihajloHoma — APPROVED** [review]

**Comment by Gondragos** [comment]
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/10939426819)
> Successfully merged:  create mode 100644 web/libs/editor/tests/integration/e2e/control_tags/classification/textarea.cy.ts

**Pull request merged**